vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HostMO.java

```diff
@@ -320,6 +320,11 @@ public AboutInfo getHostAboutInfo() throws Exception {
 
     public VmwareHostType getHostType() throws Exception {
         AboutInfo aboutInfo = getHostAboutInfo();
+        if (aboutInfo == null) {
+            String msg = "no type info about host known"
+            s_logger.error(msg)
+            throw new Exception(msg);
```

Comment:
can it be defaulted to `VmwareHostType.ESXi` ?

Reply:
what does it mean if the host does not have this info? Is it in a valid state then?

Reply:
I never face this issue before, but I know some users have faced the issue.

If we throw an Exception with message, instead of NPE, I think it does not help, as user will still get an exception.

Reply:
yes, but at least it will be clear they have an ill-configured host.

Reply:
to be honest, I do not know what is misconfigured and any other issues caused by the misconfiguration.
I think we could consider the host as an ESXi host if the about info is not found.

anyone uses ESX host or other type of host ?

Reply:
ok, if y’all insist ;)

Reply:
what I meant @weizhouapache , is your argument makes sense, I yield. Please see my changed code fragment at https://github.com/apache/cloudstack/pull/11054#discussion_r2156423875.

Reply:
agreed, assuming ESXi seems to be a good option.